.travis.yml

```diff
@@ -1,41 +1,43 @@
 language: php
 
-php:
-  - 7.1
-  - 7.2
-
 env:
   global:
-    - setup=basic
+    - SETUP=stable
 
 matrix:
+  fast_finish: true
   include:
-    - php: 7.1
-      env: setup=lowest
-    - php: 7.1
-      env: setup=stable
     - php: 7.2
-      env: setup=lowest
     - php: 7.2
-      env: setup=stable
+      env: SETUP=lowest
+    - php: 7.3
+    - php: 7.3
+      env: SETUP=lowest
 
 sudo: false
 
 cache:
   directories:
     - $HOME/.composer/cache
 
+services:
+  - memcached
+  - redis-server
+  - mysql
+
 before_install:
+  - phpenv config-rm xdebug.ini || true
   - echo "extension = memcached.so" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+  - printf "\n" | pecl install -f redis
   - travis_retry composer self-update
+  - mysql -e 'CREATE DATABASE forge;'
 
 install:
-  - if [[ $setup = 'basic' ]]; then travis_retry composer install --no-interaction --prefer-source; fi
-  - if [[ $setup = 'stable' ]]; then travis_retry composer update --prefer-source --no-interaction --prefer-stable; fi
-  - if [[ $setup = 'lowest' ]]; then travis_retry composer update --prefer-source --no-interaction --prefer-lowest --prefer-stable; fi
-
-before_script: mkdir -p build/logs
+  - if [[ $SETUP = 'stable' ]]; then travis_retry composer update --prefer-dist --no-interaction --prefer-stable --no-suggest; fi
+  - if [[ $SETUP = 'lowest' ]]; then travis_retry composer update --prefer-dist --no-interaction --prefer-lowest --prefer-stable --no-suggest; fi
```

Comment:
Why "--prefer-lowest" and "--prefer-stable"? I would think for this one you would only want to use "--prefer-lowest"

Reply:
This is essentially the exact travis config form Laravel 5.8, except versions of php & dealing with code coverage.

What is happening, is that the `SETUP` env is defaulted to `stable` on line 5, so it runs composer with whatever the latest stable versions are. Then `SETUP` is changed to `lowest`, which pulls in the oldest possible version of packages, so it is a 2x2 matrix.